source/Stargate-Examples-Tests/PetStoreAPITest.class.st

```diff
@@ -47,28 +47,176 @@ PetStoreAPITest >> testBadRequest [
 ]
 
 { #category : #'tests - CORS' }
-PetStoreAPITest >> testCORSHeadersWhenOriginIsAllowed [
+PetStoreAPITest >> testCORSAllowingAnyOrigin [
 
 	| response |
 
-	api beCORSAwareAllowing: {self baseUrl}.
+	api allowCrossOriginSharingApplying: [ :cors | cors allowAnyOrigin ].
+		
+	response := self newClient
+		url: self baseUrl / 'pets' asUrl;
+		headerAt: 'Origin' put: self baseUrl printString;
+		headerAt: 'Access-Control-Request-Headers' put: 'Content-Type';
+		options;
+		response.
+		
+	self 	
+		assert: response code equals: 204;
+		assert: ( response headers at: 'Access-Control-Allow-Origin' ) equals: '*';
```

Comment:
Maybe we can add some extensions on Hyperspace to improve accesing this headers?

Reply:
hmm the CORS headers? I added assertUrl:equals: as an utility method in other tests. You're suggesting I should add it to this tests?

Reply:
I was thinking in inject the shared pool variable into a block, like I did in the model code. But can't be an extension in this case because I need to add the shared pool